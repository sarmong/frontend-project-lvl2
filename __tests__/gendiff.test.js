import { describe, it, expect } from "@jest/globals";

import genDiff from "../src/index";

const fixturesPath = "__tests__/__fixtures__";

describe("genDiff", () => {
  it("should show correct diff for json files", () => {
    const res = genDiff(
      `${fixturesPath}/file1.json`,
      `${fixturesPath}/file2.json`
    );

    expect(res).toMatchSnapshot();
  });

  it("should show correct diff for yaml files", () => {
    const res = genDiff(
      `${fixturesPath}/file1.yaml`,
      `${fixturesPath}/file2.yaml`
    );

    expect(res).toMatchSnapshot();
  });
});
