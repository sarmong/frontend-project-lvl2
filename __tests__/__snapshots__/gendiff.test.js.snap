// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genDiff should show correct diff for json files 1`] = `
"{
    common: {
        + follow: false
          setting1: Value 1
        - setting2: 200
        - setting3: true
        + setting3: null
        + setting4: blah blah
        + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
                - wow: 
                + wow: so much
            }
              key: value
            + ops: vops
        }
    }
    group1: {
        - baz: bas
        + baz: bars
          foo: bar
        - nest: {
            key: value
        }
        + nest: str
    }
    - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
    + group3: {
        deep: {
            id: {
                number: 45
            }
        }
        fee: 100500
    }
}"
`;

exports[`genDiff should show correct diff for yaml files 1`] = `
"{
    common: {
        + follow: false
          setting1: Value 1
        - setting2: 200
        - setting3: true
        + setting3: null
        + setting4: blah blah
        + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
                - wow: null
                + wow: so much
            }
              key: value
            + ops: vops
        }
    }
    group1: {
        - baz: bas
        + baz: bars
          foo: bar
        - nest: {
            key: value
        }
        + nest: str
    }
    - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
    + group3: {
        deep: {
            id: {
                number: 45
            }
        }
        fee: 100500
    }
}"
`;
